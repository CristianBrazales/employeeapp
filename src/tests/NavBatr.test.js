import { render, screen } from "@testing-library/react";
import CustomTable from "../components/Table";
import renderer from "react-test-renderer";
import { Navbar } from "react-bootstrap";
import NavBarCustom from "../components/NavBar";
test("renders our custom nav bar empty inputs", () => {
  const component = renderer.create(
    <NavBarCustom optionsList={[]} clickHanlder={() => {}}></NavBarCustom>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.props.className).toEqual(
    "navbar navbar-expand navbar-dark bg-dark"
  );
});

test("renders our custom table with data", () => {
  let options = [{ name: "Panel", reference: "user" }];
  render(
    <NavBarCustom optionsList={options} clickHanlder={() => {}}></NavBarCustom>
  );
  let testingNode;
  testingNode = screen.getByText(/^Panel/i);
  expect(testingNode).not.toBeNull();
});
