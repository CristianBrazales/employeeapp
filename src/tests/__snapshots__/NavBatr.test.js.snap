// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders our custom nav bar empty inputs 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <div
    className="ms-5 container"
  >
    <span
      className="fw-bold fs-3 navbar-brand"
      onClick={[Function]}
    >
      Bienvenido a la nave!
    </span>
    <div
      className="me-auto navbar-nav"
      onKeyDown={[Function]}
    />
  </div>
</nav>
`;
